notebooks/search/07-inference.ipynb

```diff
@@ -0,0 +1,382 @@
+    "Read [this page](https://www.elastic.co/guide/en/elasticsearch/client/python-api/current/connecting.html#connect-self-managed-new) to learn how to connect using API keys.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "96788aa1",
+   "metadata": {},
+   "source": [
+    "## Create the inference task\n",
+    "\n",
+    "Let's create the inference task by using the [Create inference API](https://www.elastic.co/guide/en/elasticsearch/reference/current/put-inference-api.html).\n",
+    "\n",
+    "You'll need an OpenAI API key for this that you can find in your OpenAI account under the [API keys section](https://platform.openai.com/api-keys). A paid membership is required to complete the steps in this notebook as the OpenAI free trial API usage is limited."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3e6d98af",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "API_KEY = getpass.getpass('Enter OpenAI API key:  ')\n",
+    "\n",
+    "client.inference.put_model(\n",
+    "    task_type=\"text_embedding\",\n",
+    "    model_id=\"openai_embeddings\",\n",
+    "    body={\n",
+    "        \"service\": \"openai\",\n",
+    "        \"service_settings\": {\n",
+    "            \"api_key\": API_KEY\n",
+    "        },\n",
+    "        \"task_settings\": {\n",
+    "            \"model\": \"text-embedding-ada-002\"\n",
+    "        }\n",
+    "    }\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e5feaf12",
+   "metadata": {},
+   "source": [
+    "## Create an ingest pipeline with an inference processor\n",
+    "\n",
+    "Create an ingest pipeline with an inference processor by using the [`put_pipeline`](https://www.elastic.co/guide/en/elasticsearch/reference/master/put-pipeline-api.html) method. Reference the OpenAI model created above to infer against the data that is being ingested in the pipeline."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c5897fe4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "client.ingest.put_pipeline(\n",
+    "    id=\"openai_embeddings\", \n",
+    "    description=\"Ingest pipeline for OpenAI inference.\",\n",
+    "    processors=[\n",
+    "    {\n",
+    "      \"inference\": {\n",
+    "        \"model_id\": \"openai_embeddings\",\n",
+    "        \"input_output\": {\n",
+    "              \"input_field\": \"plot\",\n",
+    "              \"output_field\": \"plot_embedding\"\n",
+    "        }\n",
+    "      }\n",
+    "    }\n",
+    "  ]\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7b6dd89c",
+   "metadata": {},
+   "source": [
+    "Let's note a few important parameters from that API call:\n",
+    "\n",
+    "- `inference`: A processor that performs inference using a machine learning model.\n",
+    "- `model_id`: Specifies the ID of the machine learning model to be used. In this example, the model ID is set to `openai_embeddings`.\n",
```

Comment:
I assume for this configuration `model_id` is an arbitrary name that you pick? Maybe that needs to be clarified, if not people would be tempted to use the OpenAI model name.